packages/agent/src/oidc-v2.ts

```diff
@@ -0,0 +1,689 @@
+   * mitigate replay attacks. */
+  nonce?: string;
+  /** Custom claims. */
+  [key: string]: any;
+};
+
+/** The fields for an Web5 Connect Auth Response */
+export type Web5ConnectAuthResponse = {
+  delegation_grants: any[];
+  privateKeyJwks: Jwk[];
+};
+
+/** Represents the different OIDC endpoint types.
+ * 1. `pushedAuthorizationRequest`: client sends {@link PushedAuthRequest} receives {@link PushedAuthResponse}
+ * 2. `authorize`: provider gets the {@link HybridAuthRequest} JWT that was stored by the PAR
+ * 3. `callback`: provider sends {@link HybridAuthResponse} to this endpoint
+ * 4. `token`: client gets {@link HybridAuthResponse} from this endpoint
+ */
+type OidcEndpoint =
+  | 'pushedAuthorizationRequest'
+  | 'authorize'
+  | 'callback'
+  | 'token';
+
+/**
+ * Gets the correct OIDC endpoint out of the {@link OidcEndpoint} options provided.
+ * Handles a trailing slash on baseURL
+ *
+ * @param {Object} options the options object
+ * @param {string} options.baseURL for example `http://foo.com/connect/
+ * @param {OidcEndpoint} options.endpoint the OIDC endpoint desired
+ * @param {string} options.authParam this is the unique id which must be provided when getting the `authorize` endpoint
+ * @param {string} options.tokenParam this is the random state as b64url which must be provided with the `token` endpoint
+ */
+function buildOidcUrl({
+  baseURL,
+  endpoint,
+  authParam,
+  tokenParam,
+}: {
+  baseURL: string;
+  endpoint: OidcEndpoint;
+  authParam?: string;
+  tokenParam?: string;
+}) {
+  switch (endpoint) {
+    /** 1. client sends {@link PushedAuthRequest} & receives {@link PushedAuthResponse} */
+    case 'pushedAuthorizationRequest':
+      return appendPathToUrl({
+        path : 'par',
+        url  : baseURL,
+      });
+    /** 2. provider gets {@link HybridAuthRequest} */
+    case 'authorize':
+      return authParam
+        ? appendPathToUrl({
+          path : `authorize/${authParam}.jwt`,
+          url  : baseURL,
+        })
+        : '';
+    /** 3. provider sends {@link HybridAuthResponse} */
+    case 'callback':
+      return appendPathToUrl({
+        path : `callback`,
+        url  : baseURL,
+      });
+    /**  4. client gets {@link HybridAuthResponse */
+    case 'token':
+      return tokenParam
+        ? appendPathToUrl({
+          path : `token/${tokenParam}.jwt`,
+          url  : baseURL,
+        })
+        : '';
+    // TODO: metadata endpoints?
+    default:
+      return '';
+  }
+}
+
+/**
+ * Generates a random string value used to associate a Client authorization
+ * request with the Identity Provider's response callback.
+ *
+ * @returns A random state as a Base64Url encoded string.
+ */
+function generateRandomState() {
+  const randomStateu8a = utils.randomBytes(12);
+  const ramdomStateb64url = Convert.uint8Array(randomStateu8a).toBase64Url();
+
+  return { randomStateu8a, ramdomStateb64url };
```

Comment:
Typo `ramdom`. Also fix naming/casing:

```suggestion
  return { randomStateBytes, ramdomStateBase64Url };
```

Reply:
done